File: tests/fixtures/realworld/scala/complex/Router.scala
/** HTTP Router with middleware support. */
class Router {
  private var routes: List[Route] = List.empty

  def add(method: String, path: String, handler: () => Any): Unit = {
    routes = routes :+ Route(method, path, handler)
  }

  def dispatch(method: String, path: String): Option[Any] = {
    routes.find(r => r.method == method && r.path == path).map(_.handler())
  }
}

case class Route(method: String, path: String, handler: () => Any)

/** Logging middleware trait. */
trait Logging {
  def log(message: String): Unit = println(s"[LOG] $message")
}

/** Response status codes. */
enum StatusCode {
  case Ok, NotFound, InternalServerError
}

/** Top-level helper. */
def add(x: Int, y: Int): Int = x + y

File: tests/fixtures/realworld/dart/complex/router.dart
/// HTTP Router with middleware support.
class Router {
  final List<Route> _routes = [];
  final List<Middleware> _middleware = [];

  void get(String path, Function handler) {
    _routes.add(Route('GET', path, handler));
  }

  void post(String path, Function handler) {
    _routes.add(Route('POST', path, handler));
  }

  dynamic dispatch(String method, String path) {
    for (final route in _routes) {
      if (route.method == method && route.path == path) {
        return route.handler();
      }
    }
    throw Exception('404');
  }

  void use(Middleware mw) => _middleware.add(mw);
}

class Route {
  final String method;
  final String path;
  final Function handler;

  Route(this.method, this.path, this.handler);
}

mixin LoggerMixin {
  void log(String message) => print('[LOG] $message');
}

enum HttpMethod { get, post, put, delete, patch }

int multiply(int a, int b) => a * b;
